src/ui/components/common/filter_bar.rs

```diff
@@ -0,0 +1,43 @@
+use crate::ui::components::common::icon::{Icon, IconSize, IconType};
+use lemmy_client::lemmy_api_common::lemmy_db_schema::{ListingType, SortType};
+use leptos::prelude::*;
+use leptos_fluent::move_tr;
+use listing_type_link::ListingTypeLink;
+use sort_type_link::SortTypeLink;
+
+mod listing_type_link;
+mod sort_type_link;
+
+#[component]
+pub fn FilterBar() -> impl IntoView {
+  view! {
+    <div class="mb-4 flex flex-wrap gap-3">
+      <div class="join">
+        <button class="btn join-item btn-active">{move_tr!("posts")}</button>
```

Comment:
Random question, do these strings get compile time checked?

Reply:
If you mean checking for whether or not a class is a valid Tailwind CSS class, no. Not quite sure what you mean though.